Gemfile

```diff
@@ -2,49 +2,24 @@ source 'https://rubygems.org'
 
 ruby '3.2.2'
 
-# Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
+gem 'pg', '~> 1.1'
```

Comment:
- [ ] I can't add any items even if I re-clone the project and recreate the database  from scratch on 2 computers and using 2 browsers (Chrome and Firefox):

![image](https://github.com/sagieramos/Book-an-appointment/assets/57421823/b0a304c7-50e4-4a00-b342-34d2da7023bf)

![image](https://github.com/sagieramos/Book-an-appointment/assets/57421823/acb19da9-f38b-4143-a5e3-3e7e7518af03)

![image](https://github.com/sagieramos/Book-an-appointment/assets/57421823/49ef2413-0c93-4aff-ad2a-d14b7e307028)


If I add a debugging pry REPL in the #create method in items_controller.rb:

![image](https://github.com/sagieramos/Book-an-appointment/assets/57421823/cb9684c1-aa55-48c9-b7b6-360eae25b3fa)

![image](https://github.com/sagieramos/Book-an-appointment/assets/57421823/cac6f933-8d20-43e6-80ec-11de107ec112)

`current_user` comes as nil, and that completely breaks the server because this variable is used in many files:

![image](https://github.com/sagieramos/Book-an-appointment/assets/57421823/d61b02e1-ba35-43f1-aee2-7f14b96d8c68)

Even if I'm able to add items to the database in a hacky way (using @user in the #set_user method instead of using current_user), the rest of the app, like reservations breaks too because the server can't find out who is the `current_user`.

In order to correctly reproduce the issue, you would need to test the app as a new user.  This means deleting the database, the project folders, re-cloning the project from GitHub,  and following your own instructions to setup and run the project.  Due to local/session storage, the frontend app should be tested on an incognito tab, and that way you will have the same environment as a new user.

I would debug as a new user and take a look at what's going on before going into the final evaluation (most likely something to do with the authorization and/or the creation of the `current_user` variable, as shown above). 👍 

### Important:

If you are confident that the app works well, and since this is a special review cycle designed to help you in the final test, you can ignore my suggestions, merge, submit the project as is, and go straight into the final evaluation.

Reply:
Thank you for thoroughly reviewing this project. I did not encounter any errors after recreating my database. Here is a video record: 

[Screencast from 02-02-2024 01:02:58.webm](https://github.com/sagieramos/Book-an-appointment/assets/31627384/79140475-922d-4d42-9f82-122acfb833df)